### PR TITLE
Replace redundant methods in stats

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR refactors the cg/apps/cgstats/crud directory, removing the get_project_id and get_sample_id functions in find.py and updating the code to use more suitable functions, improving code quality and readability.

Even if the end goal is to bin this app, it makes it easier to work with now.

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/apps/cgstats/crud/create.py [hunk 1]
```diff
@@ -198,39 +198,39 @@ def _calculate_read_counts(demux_sample: DragenDemuxSample) -> int:
 def create_projects(manager: StatsAPI, project_names: Iterable[str]) -> Dict[str, int]:
```
Comment (diitaz93) [inline cg/apps/cgstats/crud/create.py:198]: docstring

File: cg/apps/cgstats/crud/create.py [hunk 2]
```diff
@@ -198,39 +198,39 @@ def _calculate_read_counts(demux_sample: DragenDemuxSample) -> int:
...
-            LOG.info("Project %s already exists", project_name)
-        project_name_to_id[project_name] = project_id
+        project: Optional[Project] = get_or_create_project(
+            manager=manager, project_name=project_name
+        )
+        project_name_to_id[project_name] = project.project_id
     return project_name_to_id
 
 
 def _create_samples(
```
Comment (diitaz93) [inline cg/apps/cgstats/crud/create.py:208]: Maybe if it creates only one sample is better to rename it to `_create_sample`

File: cg/apps/cgstats/crud/create.py [hunk 3]
```diff
@@ -198,39 +198,39 @@ def _calculate_read_counts(demux_sample: DragenDemuxSample) -> int:
...
     manager: StatsAPI, sample: NovaSeqSample, project_name_to_id: Dict[str, int]
-) -> Union[int, None]:
-    """handles sample objects creation for the table `Sample` in cgstats"""
+) -> Optional[int]:
+    """Create a new Sample object in the cgstats database if it doesn't already exist."""
+
+    if sample.project == "indexcheck":
+        LOG.debug("Skip adding indexcheck sample to database")
```
Comment (diitaz93) [inline cg/apps/cgstats/crud/create.py:214]: Maybe this should be a `LOG.warning`or `LOG.info`

File: cg/apps/cgstats/crud/create.py [hunk 4]
```diff
@@ -198,39 +198,39 @@ def _calculate_read_counts(demux_sample: DragenDemuxSample) -> int:
...
+        project_name_to_id[project_name] = project.project_id
     return project_name_to_id
 
 
 def _create_samples(
     manager: StatsAPI, sample: NovaSeqSample, project_name_to_id: Dict[str, int]
-) -> Union[int, None]:
-    """handles sample objects creation for the table `Sample` in cgstats"""
+) -> Optional[int]:
+    """Create a new Sample object in the cgstats database if it doesn't already exist."""
```
Comment (diitaz93) [inline cg/apps/cgstats/crud/create.py:211]: update doctring with "also returns sample id"?

**Review by diitaz93 — APPROVED** [review]
Looks good! Left some comments despite the module going to be depreciated 😅 Maybe there si a better name for functions starting with `get_or_create`?

File: cg/apps/cgstats/crud/create.py (hunk 3 above)
Comment (seallard) [reply cg/apps/cgstats/crud/create.py:214]: I'm keeping it as a debug since that was how it was before, just refactoring it.

**Pull request merged**